### PR TITLE
Feature/3119 drawer footer mui style engine

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes # BLUI-3119

DrawerFooter 
Changes proposed in this Pull Request:
Update styles to use Mui-v5 Styled components
Update component to extend sx prop

Test:
There is a showcase branch feature/3050-channel-value-mui-v5-styles-update that you can use to test this.
When testing, ensure that there are no breaking changes and that the old API properties still work the same and that you can still style things with inline styles as well as using our classes prop.
Also, test out the new sx prop extension. You can apply styles to the root by using sx directly on the component as well as by targeting nested items by the generated classnames.

### CODE REVIEW

**Comment by emclaug2** [comment]
Screen Shot 2022-05-05 at 8 58 47 AM

@bkarambe were you able to get the showcase working with the `sx` prop applied to the footer?  I'm encountering the same error.

File: components/src/core/Drawer/DrawerFooter/DrawerFooter.tsx [hunk 1]
```diff
@@ -76,9 +75,9 @@ const DrawerFooterRender: React.ForwardRefRenderFunction<unknown, DrawerFooterPr
     } = props;
     const { open: drawerOpen = true } = useDrawerContext();
     return (
-        <Root>
+        <Fragment>
```
Comment (daileytj) [inline components/src/core/Drawer/DrawerFooter/DrawerFooter.tsx:78]: I think I would prefer to pass sx through to the Root in this situation.

You can add this as a prop:
````sx?: SxProps;```

**Comment by bkarambe** [comment]
> Screen Shot 2022-05-05 at 8 58 47 AM
> 
> @bkarambe were you able to get the showcase working with the `sx` prop applied to the footer? I'm encountering the same error.

Yes I tried applying sx prop and it was working fine.

**Comment by jeffvg** [comment]
yarn lint in CI fails - FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

running node -e 'console.log(v8.getHeapStatistics().heap_size_limit/(1024*1024))' in root of components dir reports (4144)
but not certain if node needs to have memory bumped up or something else

File: components/src/core/Drawer/DrawerFooter/DrawerFooter.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import React from 'react';
```
Comment (emclaug2) [inline components/src/core/Drawer/DrawerFooter/DrawerFooter.tsx:1]: There is something in this file that is causing an issue with linting.  Linting takes 4x more time whenever this file is included.  I played around with this for some time & cannot figure out what the culprit is.  I suggest making an issue for this & temporarily adding this file to the eslint ignore list. (see commit)

**Comment by bkarambe** [comment]
> Passes lint, but now unit tests are acting up.

Enzyme is not compatible with react 18. We should use react testing library. https://dev.to/wojtekmaj/enzyme-is-dead-now-what-ekl

File: components/src/core/Drawer/DrawerFooter/DrawerFooter.tsx (hunk 2 above)
Comment (bkarambe) [reply components/src/core/Drawer/DrawerFooter/DrawerFooter.tsx:1]: Issue created https://github.com/brightlayer-ui/react-component-library/issues/433

**Comment by emclaug2** [comment]
There's something happening here that's not being understood.  Linting is taking 4x as long, unit tests are timing out in the circle ci.  This cannot get merged in until we've figured out why these things are happening.

If you lint the components folder using the `--debug` flag to eslint, we can see that it's taking me 27 seconds (on my desktop) to lint the DrawerFooter file.

```
  eslint:file-enumerator Yield: DrawerFooter.tsx +3ms
  eslintrc:cascading-config-array-factory Load config files for C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter. +2ms
  eslintrc:cascading-config-array-factory Cache hit: C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter. +0ms
  eslint:cli-engine Lint C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter\DrawerFooter.tsx +0ms
  eslint:linter Linting code for C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter\DrawerFooter.tsx (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter\DrawerFooter.tsx +1ms
  eslint:linter Parsing: C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter\DrawerFooter.tsx +2ms
  eslint:linter Parsing successful: C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter\DrawerFooter.tsx +2s
  eslint:linter Scope analysis: C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter\DrawerFooter.tsx +1ms
  eslint:linter Scope analysis successful: C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter\DrawerFooter.tsx +0ms
  eslint:config-comment-parser Parsing list config +0ms
  eslint:config-comment-parser Parsing list config +1ms
  eslint:linter Generating fixed text for C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter\DrawerFooter.tsx (pass 1) +25s
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:file-enumerator Didn't match: DrawerFooterClasses.tsx +27s
  eslint:file-enumerator Didn't match: index.ts +1ms
  eslint:file-enumerator Leave the directory: C:\Users\Evan\Documents\git\pxblue\react-component-library\components\src\core\Drawer\DrawerFooter +0ms
  eslint:file-enumerator Didn't match: DrawerHeader.tsx +0ms
  eslintrc:ignore-pattern Check {
  filePath: 'C:\\Users\\Evan\\Documents\\git\\pxblue\\react-component-library\\components\\src\\core\\Drawer\\DrawerNavGroup\\',
  dot: false,
  relativePath: 'src/core/Drawer/DrawerNavGroup/',
  result: false
} +27s
```

**Review by emclaug2 — APPROVED** [review]

**Pull request merged**